src/datasets/arrow_dataset.py

```diff
@@ -3522,9 +3526,12 @@ def push_to_hub(
             branch (Optional :obj:`str`):
                 The git branch on which to push the dataset. This defaults to the default branch as specified
                 in your repository, which defaults to `"main"`.
+            max_shard_size (`int` or `str`, *optional*, defaults to `"500MB"`):
+                The maximum size of the dataset shards to be uploaded to the hub. If expressed as a string, needs to be digits followed by a unit
+                (like `"5MB"`).
             shard_size (Optional :obj:`int`):
-                The size of the dataset shards to be uploaded to the hub. The dataset will be pushed in files
-                of the size specified here, in bytes. Defaults to a shard size of 500MB.
+                .. deprecated:: 2.1.1
+                    'shard_size' was renamed to 'max_shard_size' in version 2.1.1 and will be removed in 2.4.0.
```

Comment:
The rendering in the doc of this syntax is not perfect. I don't think we have the equivalent for markdown rendering, so I guess we can just write it in plain text

<img width="873" alt="image" src="https://user-images.githubusercontent.com/42851186/164684552-b5416aab-0b30-46ec-bf5f-7e2480b3645f.png">

```suggestion
                Deprecated: 'shard_size' was renamed to 'max_shard_size' in version 2.1.1 and will be removed in 2.4.0.
```